python/pyspark/worker.py

```diff
@@ -841,6 +845,63 @@ def _remove_partition_by_exprs(self, arg: Any) -> Any:
             "the query again."
         )
 
+    # Compares each UDTF output row against the output schema for this particular UDTF call,
+    # raising an error if the two are incompatible.
+    def check_output_row_against_schema(row: Any) -> None:
```

Comment:
@ueshin do you think this will add extra performance overhead if we check this for each output row?

Reply:
Yes, this will add huge performance overhead.

@dtenedor Could we at least build the check function based on the data type in advance?

```py
check_output_row_against_schema = _build_null_checker(return_type)
```

Checking the data type and `nullable` each row should be too expensive.

The builder should be placed somewhere reusable.

Reply:
Sure, I added these check back for onw.